File: Data Structures & Algorithms/house-robber-ii/submission-4.cpp
class Solution {
public:
    int go(vector<int>&nums,int lidx, int ridx){
        if(ridx-lidx<=1) return max(nums[lidx], nums[ridx]);
        vector<int> dp(ridx-lidx+1);
        dp[0]= nums[lidx]; 
        dp[1]=max(nums[lidx],nums[lidx+1]);
        for(int i=2;i<ridx-lidx+1;++i){
            dp[i]= max(nums[i+lidx]+dp[i-2], dp[i-1]);
        }
        return dp[ridx-lidx];
    }
    int rob(vector<int>& nums) {
        if(nums.size()<=2) return *max_element(nums.begin(),nums.end());
        return max(go(nums,0,nums.size()-2),go(nums,1,nums.size()-1));
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        ans, temp_ans, nums_set = 0,0, set(nums)
        for x in nums_set:
            if x+1 not in nums_set:
                temp_num, temp_ans = x, 1
                while temp_num -1 in nums_set:
                    temp_num -=1
                    temp_ans += 1
                ans = max(ans, temp_ans)
        return ans
